dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/FilterJdbcPredicate.java

```diff
@@ -240,6 +248,50 @@ private static String rightOperandSql(QueryOperator operator, Parameter paramete
     return " :" + name;
   }
 
+  @Nonnull
+  private static String multiTextClause(
+      String column, QueryOperator operator, Parameter parameter) {
+    String unnestSql = "unnest(string_to_array(lower(" + column + "), ',')) AS val";
+    String trimmed = "trim(val)";
+    String param = parameter != null ? parameter.name() : null;
+
+    return switch (operator) {
+      case IEQ, EQ, IN ->
+          String.format("exists (select 1 from %s where %s in (:%s))", unnestSql, trimmed, param);
+      case NE, NEQ ->
+          String.format(
+              "not exists (select 1 from %s where %s in (:%s))", unnestSql, trimmed, param);
+      case LIKE, ILIKE ->
+          String.format("exists (select 1 from %s where %s like :%s)", unnestSql, trimmed, param);
+      case NLIKE, NILIKE ->
+          String.format(
+              "not exists (select 1 from %s where %s like :%s)", unnestSql, trimmed, param);
+      case SW ->
+          String.format(
+              "exists (select 1 from %s where %s like CONCAT(:%s, '%%'))",
+              unnestSql, trimmed, param);
+      case EW ->
+          String.format(
+              "exists (select 1 from %s where %s like CONCAT('%%', :%s))",
+              unnestSql, trimmed, param);
+      case NULL ->
```

Comment:
`NULL` and `NNULL` case follow the same logic as a data value that is not multitext, right?
Can we make sure we use the same logic? If we have an empty string for a `TEXT` data value and we use the `NULL` operator, what is the outcome there?

Reply:
If a data value is stored as an empty string `("")`, then using the filter `filter=de_uid:null` will retrieve this event. In contrast, the filter `filter=de_uid:!null` will retrieve all events except the one with the empty string.
This case is covered in one of the tests as well.